internal/impl/packages.go

```diff
@@ -244,7 +252,48 @@ func (d *Devbox) ensureNixpkgsPrefetched() error {
 	}
 	fmt.Fprintf(d.writer, "Ensuring nixpkgs registry is downloaded: ")
 	color.New(color.FgGreen).Fprintf(d.writer, "Success\n")
-	return nil
+
+	return d.saveToNixpkgsCommitFile()
+}
+
+func (d *Devbox) isNixpkgsFetched() (bool, error) {
+	path := d.nixpkgsCommitFilePath()
+	if !fileutil.Exists(path) {
+		return false, nil
+	}
+
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return false, errors.WithStack(err)
+	}
+	for _, line := range strings.Split(string(contents), "\r\n") {
+		if line == d.cfg.Nixpkgs.Commit {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (d *Devbox) saveToNixpkgsCommitFile() error {
+	path := d.nixpkgsCommitFilePath()
+
+	if err := os.Mkdir(filepath.Dir(path), 0755); err != nil && !errors.Is(err, fs.ErrExist) {
+		return errors.WithStack(err)
+	}
+
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer file.Close()
+
+	_, err = fmt.Fprintf(file, "%s", d.cfg.Nixpkgs.Commit)
```

Comment:
The approach I had in mind was a bit different. Instead of saving only the nix commit (which doesn't guarantee that the nixpackges is actually downloaded) I was thinking of saving a map:

commit -> directory

by saving this we can check the directory actually exists. And we can make it a bit more general by calling file `nixpkgs.json`

and having:

```json
"commit_map" : {
  "hash-123": "/nix/store/..."
}
```

Reply:
oh that's a good idea. It'll get rid of the Risk I outline in the PR summary.